content/en/blog/tracing-syscalls-using-ebpf/index.md

```diff
@@ -0,0 +1,278 @@
+       src="images/ebpf-11.png">
+  </img>
+</a><br><br>
+
+### bpf_program__attach_tracepoint
+
+This function is used to attach an eBPF program to a kernel tracepoint.
+
+<br><a target="_blank" href="images/ebpf-12.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Function which is used to attach an eBPF program to a kernel tracepoint."
+       src="images/ebpf-12.png">
+  </img>
+</a><br><br>
+
+### ring_buffer__new
+
+This function is used for creating and opening a new ringbuf manager.
+
+<br><a target="_blank" href="images/ebpf-13.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Function which is used for creating and opening a new ringbuf manager."
+       src="images/ebpf-13.png">
+  </img>
+</a><br><br>
+
+### ring_buf__consume
+
+Used to remove or consume data from a ring buffer.
+
+<br><a target="_blank" href="images/ebpf-14.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Program used to remove or consume data from a ring buffer."
+       src="images/ebpf-14.png">
+  </img>
+</a><br><br>
+
+## BTF (BPF Type Format)
+
+It provides a way to describe the types of data structures used by eBPF programs, allowing for improved type safety, debugging, and introspection.
+
+Now, let's write a program that sends data to userspace using ringbuf.
+
+<br><a target="_blank" href="images/ebpf-15.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Program that sends data to userspace using ringbuf"
+       src="images/ebpf-15.png">
+  </img>
+</a><br><br>
+
+Having created the program, we can write a loader to load this eBPF program.
+
+<br><a target="_blank" href="images/ebpf-16.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="A loader to load this eBPF program."
+       src="images/ebpf-16.png">
+  </img>
+</a><br><br>
+
+
+The infinite loop is necessary to ensure that the program continuously checks for new events in the ring buffer. Without the loop, the program would only consume events that were already in the buffer at the time of the initial `ring_buffer__consume()` call. By looping and calling `ring_buffer__consume()` repeatedly, the program can retrieve events as soon as they become available and process them in real-time. The `sleep(1)` call within the loop serves to reduce the CPU usage of the program by introducing a one-second delay between each call to `ring_buffer__consume()`.
+
+<br><a target="_blank" href="images/ebpf-17.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Commands to compile and run the above program."
+       src="images/ebpf-17.png">
+  </img>
+</a><br><br>
+
+<br><a target="_blank" href="images/ebpf-res-02.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Results after compiling and executing the above program."
+       src="images/ebpf-res-02.png">
+  </img>
+</a><br><br>
+
+Great, we were able to recover the process name as well as the PID!
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
```suggestion
```

Reply:
Can you also add a conclusion to introduce the part 2.

Reply:
done

Reply:
done